enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,851 @@
+
+### User Stories
+
+- As an OpenShift user, I want to have an option to dynamically enable `istio-csr`, so that it can be used only
+  when required by creating the custom resource.
+- As an OpenShift user, I want to have an option to dynamically configure `istio-csr`, so that only the required
+  features can be enabled by updating the custom resource.
+- As an OpenShift user, I should be able to disable `istio-csr` when not required by removing the custom resource,
+  and controller should cleanup all resources created for the `istio-csr` deployment.
+- As an OpenShift user, I should be able to install `istio-csr` in the upgrade cluster where istiod control plane
+  is active and should be able to update the certificate endpoint to `istio-csr` endpoint.
+- As an OpenShift user, I want to have an option to dynamically enable monitoring for the `istio-csr` project and
+  to use the OpenShift monitoring solution when required.
+
+### Goals
+
+- `cert-manager-operator` to be extended to manage `istio-csr` along with currently managed `cert-manager`.
+- New custom resource `istiocsr` to be made available to enable, configure and disable deployment.
+
+### Non-Goals
+
+- `istio-csr` can be used only with supported version of `OpenShift Service Mesh`. Please refer `Version Skew Strategy`
+  section for more details.
+
+## Proposal
+
+`istio-csr` will be managed by `cert-manager-operator` which can be enabled and configured by a user as and
+when required, using new custom resource which will be made available for configuring `istio-csr`.
+
+A new controller will be added to `cert-manager-operator` to manage and maintain the `istio-csr` deployment in
+desired state which can be monitored by user through the status sub-resource of the new custom resource.
+
+`istio-csr` controller will make use of static manifest templates for creating below resources required for
+successfully deploying `istio-csr` agent. Please refer `Implementation Details/Notes/Constraints` section for more details.
+* `certificate.cert-manager.io`
+* `clusterissuer.cert-manager.io`
+* `issuer.cert-manager.io`
+* `clusterrole.rbac.authorization.k8s.io`
+* `clusterrolebinding.rbac.authorization.k8s.io`
+* `deployment`
+* `service`
+* `servicemonitor.monitoring.coreos.com`
+* `namespace`
+* `role.rbac.authorization.k8s.io`
+* `rolebinding.rbac.authorization.k8s.io`
+* `serviceaccount`
+
+`istio-csr` CR object will be a singleton object, must be named `default` and an OpenShift user will have to
+create the same for enabling the deployment. `istio-csr` will be deployed in the namespace where the CR object is created.
+
+When `istiocsr` custom resource is created without defining `issuerref`, the operator will create a `ClusterIssuer` in the
```

Comment:
The usage of terms `istiocsr` and `istio-csr` is not consistent across the proposal. Also, a section that defines these terms will help to alleviate the ambiguity.